library/std/src/sys/wasi/thread.rs

```diff
@@ -62,7 +162,17 @@ impl Thread {
     }
 
     pub fn join(self) {
-        self.0
+        cfg_if::cfg_if! {
+            if #[cfg(target_feature = "atomics")] {
+                unsafe {
+                    let ret = libc::pthread_join(self.id, ptr::null_mut());
+                    mem::forget(self);
+                    assert!(ret == 0, "failed to join thread: {}", io::Error::from_raw_os_error(ret));
```

Comment:
Not super familiar with Rust so not 100% sure what the `assert()` does - I guess it `panic`s - if so, should we rather return `Err()` here?

Reply:
I'd prefer that, now it also seems to require change of the signature as in https://github.com/rust-lang/rust/pull/112922#discussion_r1250580223

Reply:
I agree with @bjorn3, this should use `rtabort!()`.

Reply:
I'm not sure I follow

I thought @bjorn3 suggested to use `rtabort!()` instead of unwrap in different file https://github.com/rust-lang/rust/issues/112521#issuecomment-1587261695 ?

Or do I just need to replace `assert!` with `rtabort!` ?

Reply:
I updated it and checked that the tests still pass
Is that what you had in mind?

Reply:
Yeah, this is what I had in mind.